compose-app/src/main/java/com/mapbox/maps/extension/compose/MapboxMap.kt

```diff
@@ -0,0 +1,185 @@
+package com.mapbox.maps.extension.compose
+
+import android.annotation.SuppressLint
+import androidx.compose.runtime.*
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.LocalInspectionMode
+import androidx.compose.ui.viewinterop.AndroidView
+import androidx.core.content.res.ResourcesCompat
+import com.mapbox.maps.*
+import com.mapbox.maps.extension.compose.viewport.MapViewport
+import com.mapbox.maps.extension.style.StyleContract
+import com.mapbox.maps.extension.style.style
+import com.mapbox.maps.plugin.LocationPuck2D
+import com.mapbox.maps.plugin.delegates.MapPluginProviderDelegate
+import com.mapbox.maps.plugin.gestures.OnMapClickListener
+import com.mapbox.maps.plugin.gestures.OnMapLongClickListener
+import com.mapbox.maps.plugin.gestures.generated.GesturesSettings
+import com.mapbox.maps.plugin.gestures.gestures
+import com.mapbox.maps.plugin.locationcomponent.R
+import com.mapbox.maps.plugin.locationcomponent.generated.LocationComponentSettings
+import com.mapbox.maps.plugin.locationcomponent.location
+import com.mapbox.maps.plugin.viewport.ViewportStatusObserver
+import com.mapbox.maps.plugin.viewport.viewport
+
+internal class MapRootNode(
+  val mapController: MapControllable,
+  val mapPluginProvider: MapPluginProviderDelegate,
+  var gesturesState: GesturesSettings,
+  var styleState: StyleContract.StyleExtension,
+  var cameraOptions: CameraOptions
+) : MapNode {
+
+  override fun onAttached() {
+    // setup map listeners
+  }
+}
+
+
+@SuppressLint("Lifecycle")
+@Composable
+@MapboxMapComposable
+public fun MapboxMap(
+  modifier: Modifier = Modifier,
+  mapInitOptions: MapInitOptions = MapInitOptions(LocalContext.current),
```

Comment:
Might be better to pass the app context?
```suggestion
  mapInitOptions: MapInitOptions = MapInitOptions(LocalContext.current.applicationContext),
```